samza-api/src/main/java/org/apache/samza/system/IncomingMessageEnvelope.java

```diff
@@ -90,17 +90,6 @@ public boolean isEndOfStream() {
     return END_OF_STREAM_OFFSET.equals(offset);
   }
 
-  /**
-   * Builds an end-of-stream envelope for an SSP. This is used by a {@link SystemConsumer} implementation to
-   * indicate that it is at end-of-stream. The end-of-stream message should not delivered to the task implementation.
-   *
-   * @param ssp The SSP that is at end-of-stream.
-   * @return an IncomingMessageEnvelope corresponding to end-of-stream for that SSP.
-   */
-  public static IncomingMessageEnvelope buildEndOfStreamEnvelope(SystemStreamPartition ssp) {
```

Comment:
This is not backwards compatible (I'm using it in Beam, BTW). Can we \@Deprecate with an indication of the preferred alternative?